### PR TITLE
Introduce insightsoperators.openshift.io CR & implement its gather st…

### PR DESCRIPTION
**Pull request by tremes**
…atus attribute

This add a new CR for the ` insightsoperators.openshift.io` resource and implements its `gatherStatus` part.

## Categories

- [ ] Bugfix
- [ ] Enhancement
- [ ] Backporting
- [X] Others (CI, Infrastructure, Documentation)

## Sample Archive

No new data

## Unit Tests

extended a bit
- `pkg/controller/periodic/periodic_test.go`

## Privacy

Yes. There are no sensitive data in the newly collected information.

## Breaking Changes

No

## References

https://issues.redhat.com/browse/CCXDEV-8885
https://access.redhat.com/solutions/???

### CODE REVIEW

**Comment by wking** [comment]
You probably want to extend `relatedObjects` so `must-gather` and friends pick these up.  [Currently][1]:

```console
$ curl -s https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_insights-operator/666/pull-ci-openshift-insights-operator-master-e2e/1562059761355067392/artifacts/e2e/gather-extra/artifacts/clusteroperators.json | jq -c '.items[] | select(.metadata.name == "insights").status.relatedObjects[]'
{"group":"","name":"openshift-insights","resource":"namespaces"}
{"group":"apps","name":"insights-operator","namespace":"openshift-insights","resource":"deployments"}
{"group":"","name":"pull-secret","namespace":"openshift-config","resource":"secrets"}
{"group":"","name":"support","namespace":"openshift-config","resource":"secrets"}
{"group":"","name":"gather","namespace":"openshift-insights","resource":"serviceaccounts"}
{"group":"","name":"operator","namespace":"openshift-insights","resource":"serviceaccounts"}
{"group":"","name":"metrics","namespace":"openshift-insights","resource":"services"}
{"group":"","name":"service-ca-bundle","namespace":"openshift-insights","resource":"configmaps"}
```

And:

```console
$ curl -s https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_insights-operator/666/pull-ci-openshift-insights-operator-master-e2e/1562059761355067392/artifacts/e2e/gather-must-gather/artifacts/must-gather.tar | tar -tvz | grep insightsoperator
-rw------- 1008750000/root   19757 2022-08-23 07:53 registry-build04-ci-openshift-org-ci-op-rr1gwtx2-stable-sha256-7ba86eeb5e5e97946caaff9c94f1e1350c91b8773cf49bbfc9d6a36ab4f47280/cluster-scoped-resources/apiextensions.k8s.io/customresourcedefinitions/insightsoperators.operator.openshift.io.yaml
-rw------- 1008750000/root    5242 2022-08-23 07:53 registry-build04-ci-openshift-org-ci-op-rr1gwtx2-stable-sha256-7ba86eeb5e5e97946caaff9c94f1e1350c91b8773cf49bbfc9d6a36ab4f47280/cluster-scoped-resources/apiserver.openshift.io/apirequestcounts/insightsoperators.v1.operator.openshift.io.yaml
```

[1]: https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_insights-operator/666/pull-ci-openshift-insights-operator-master-e2e/1562059761355067392

File: manifests/09-operator-cr.yaml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+apiVersion: operator.openshift.io/v1
+kind: InsightsOperator
+metadata:
+  name: cluster
```
Comment (wking) [inline manifests/09-operator-cr.yaml:4]: You need to annotate this with `include.release.openshift.io/...` for the cluster profiles it should belong to.  See your other manifests for examples.  [As it stands now][1]:

```console
$ curl -s https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_insights-operator/666/pull-ci-openshift-insights-operator-master-e2e/1562059761355067392/artifacts/e2e/gather-extra/artifacts/pods/openshift-cluster-version_cluster-version-operator-5bff5b8bd6-2d6n7_cluster-version-operator.log | grep kind=InsightsOperator
I0823 13:42:20.564370       1 payload.go:210] excluding 0000_50_insights-operator_09-operator-cr.yaml group=operator.openshift.io kind=InsightsOperator namespace= name=cluster: no annotations
I0823 13:42:21.332159       1 payload.go:210] excluding 0000_50_insights-operator_09-operator-cr.yaml group=operator.openshift.io kind=InsightsOperator namespace= name=cluster: no annotations
```

[1]: https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_insights-operator/666/pull-ci-openshift-insights-operator-master-e2e/1562059761355067392

Comment (tremes) [reply manifests/09-operator-cr.yaml:4]: Thank you

**Review by rluders — APPROVED** [review]
I tested yesterday.
/lgtm

**Pull request merged**